app/services/proofing/resolution/plugins/aamva_plugin.rb

```diff
@@ -0,0 +1,133 @@
+# frozen_string_literal: true
+
+module Proofing
+  module Resolution
+    module Plugins
+      class AamvaPlugin
+        SECONDARY_ID_ADDRESS_MAP = {
+          identity_doc_address1: :address1,
+          identity_doc_address2: :address2,
+          identity_doc_city: :city,
+          identity_doc_address_state: :state,
+          identity_doc_zipcode: :zipcode,
+        }.freeze
+
+        def call(
+          applicant_pii:,
+          current_sp:,
+          instant_verify_result:,
+          ipp_enrollment_in_progress:,
+          timer:
+        )
+          should_proof = should_proof_state_id_with_aamva?(
+            applicant_pii:,
+            instant_verify_result:,
+            ipp_enrollment_in_progress:,
+          )
+
+          if !should_proof
+            return out_of_aamva_jurisdiction_result
+          end
+
+          applicant_pii_with_state_id_address =
+            if ipp_enrollment_in_progress
+              with_state_id_address(applicant_pii)
+            else
+              applicant_pii
+            end
+
+          timer.time('state_id') do
+            proofer.proof(applicant_pii_with_state_id_address)
+          end.tap do |result|
+            if result.exception.blank?
+              Db::SpCost::AddSpCost.call(
+                current_sp,
+                :aamva,
+                transaction_id: result.transaction_id,
+              )
+            end
+          end
+        end
+
+        def aamva_supports_state_id_jurisdiction?(applicant_pii)
+          state_id_jurisdiction = applicant_pii[:state_id_jurisdiction]
+          IdentityConfig.store.aamva_supported_jurisdictions.include?(state_id_jurisdiction)
```

Comment:
Nothing to change but while testing I wondered why we collect issuing state and address on ID.  They should always be the same. I will ask my team why we have both.

Reply:
It's counterintuitive, but definitely possible! Think maybe you get a license issued to you, then you move to a new state. You then inform the old state's DMV about your new address.